lib/commands/connect/shared.js

```diff
@@ -109,77 +110,69 @@ let request = exports.request = function(token, method, path, data){
   return promise;
 };
 
-let withUserConnections = exports.withUserConnections = function(token, appName, flags, allowNone, heroku){
+let withUserConnections = exports.withUserConnections = co.wrap(function* (token, appName, flags, allowNone, heroku){
   global_config();
 
   let resource_id = null;
   if (flags && flags.resource) {
     console.log("Calling Platform API for resouce id");
-    return Q.Promise(function(resolve, reject) {
-      return heroku.apps(appName).addons(flags.resource).info(function(err, addon) {
-        if(err) {
-          if (err.statusCode === 404) {
-            console.log("Connection with resource name '" + flags.resource + "' not found");
-            reject();
-          } else {
-            console.log(err);
-            reject(err);
-          }
+
+    return co(function* () {
+      try {
+        let addon = yield heroku.apps(appName).addons(flags.resource).info();
+      } catch(err) {
+        if (err.statusCode === 404) {
+          return yield Promise.reject("Connection with resource name '" + flags.resource + "' not found");
```

Comment:
Yeah, after your comment I tried that as well, because it tracked with other behavior I had seen. Sure enough, it seems to work fine. I'd *much* rather just throw errors directly, because it just simplifies how everything looks and removes extra things to think about that don't really matter.